docs/configuring-ingest.asciidoc

```diff
@@ -0,0 +1,87 @@
+// This file was copied over from libbeat and
+// then adapted to APM Server specific needs
+
+[[configuring-ingest-node]]
+== Parse data using ingest node pipelines
+
+When you use Elasticsearch for output, you can configure {beatname_uc} to use
+{elasticsearch}/ingest.html[ingest node] to pre-process documents before the
+actual indexing takes place in Elasticsearch. Ingest node is a convenient
+processing option when you want to do some extra processing on your data, but
+you do not require the full power of Logstash. For example, you can create an
+ingest node pipeline in Elasticsearch that consists of one processor that
+removes a field in a document followed by another processor that renames a
+field.
+
+Using pipelines involves two steps.
+First you need to <<register-pipelines,register a pipeline>> in Elasticsearch.
+Then the pipeline needs to be <<apply-pipelines, applied during data ingestion>>.
+
+[[register-pipelines]]
+[float]
+=== Registering Pipelines in Elasticsearch
+For registering a pipeline in Elasticsearch you can either manually upload
+a pipeline definition or you can configure APM Server to register pipelines on startup.
+
+[[register-pipelines-apm-server]]
+[float]
+==== Register pipelines on APM Server startup
+Automatic pipeline registration requires `output.elasticsearch` to be enabled and configured,
+as well as having the required Elasticsearch Processor Plugins installed.
+
+APM Server default pipelines require you to have the Ingest User Agent Plugin installed.
+Find the default pipeline configuration at `ingest/pipeline/definition.json` of your APM Server
+installation's home directory.
+In case you want to add, change or remove pipelines in Elasticsearch,
+you can simply change the definitions in this file
+and restart your APM Server or run `apm-server setup --pipeline`
+
+By default pipeline registration is disabled,
+you need to set `apm-server.register.ingest.pipeline` to true to enable.
+See available configuration options for <<register.ingest.pipeline ,pipeline registration>>.
```

Comment:
Alternative:
"See how to configure <<register.ingest.pipeline ,pipeline registration>>."